Generator/Sources/NeedleFramework/Generating/Serializers/DependencyProviderClassNameSerializer.swift

```diff
@@ -38,3 +38,27 @@ class DependencyProviderClassNameSerializer: Serializer {
 
     private let provider: ProcessedDependencyProvider
 }
+
+/// A serializer that produces the class name for the dependency provider base class.
+class DependencyProviderBaseClassNameSerializer: Serializer {
```

Comment:
Might be helpful to make these `final`, since it might be confusing that these classes have the word `Base` in it?

Reply:
sure, it's serializer is for the base class name, but I can see how "base" in there can be confusing.